source/core/GenericFunctions.cpp

```diff
@@ -679,36 +680,43 @@ InstructionCore* EmitMaxMinEltsInstruction(ClumpParseState* pInstructionBuilder)
     }
     return pInstruction;
 }
+
 //------------------------------------------------------------
-static bool GetMinimumArrayDimensions(TypedArrayCoreRef sourceArray1, TypedArrayCoreRef sourceArray2,
+static bool GetMinimumArrayDimensions(std::vector<TypedArrayCoreRef> arrays,
     ArrayDimensionVector* newDimensionLengths, IntIndex* newRank)
 {
-    VIREO_ASSERT(sourceArray1 != null);
-    VIREO_ASSERT(sourceArray2 != null);
+    int arrayCount = arrays.size();
+    for (int i = 0; i < arrayCount; i++) {
+        VIREO_ASSERT(sourceArrays[i] != null);
```

Comment:
I think sourceArrays is meant to be arrays. I wonder why you didn't get a compile error.